operator/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -192,6 +198,45 @@ type ClusterSpec struct {
 	NodePools []NodePoolSpec `json:"nodePools,omitempty"`
 }
 
+// ClusterConfiguration holds values (or references to values) that should be used
+// to configure the cluster. Where the cluster schema defines a non-string type for a
+// given key, the corresponding values here should be string-encoded (according to yaml
+// rules)
+// TODO: this type should be lifted to a shared module rather than duplicated in the `redpanda` CRD definition.
+type ClusterConfiguration struct {
```

Comment:
Do we expect there to be members other than `Values` here? If not, could this just be `map[string]ClusterConfigValue`?

Reply:
I think probably having an `",omitempty"` annotation on this is appropriate - it'd let us come back and slide in typed fields later (issues with the generation tooling require monomorphic types notwithstanding).